lib/validators/properties-validator.ts

```diff
@@ -88,4 +91,17 @@ export class PropertiesValidator extends Validator {
             }
         }
     }
+
+    private validateChildProperties(
+        property: ComponentProperty,
+        componentPropertyNames: Set<string>,
+        component: ParsedComponent,
+    ) {
+        (property.childProperties || []).forEach((conditionalChildProperties) => {
+            const conditionalUsedProperyNames = new Set(componentPropertyNames);
+            (conditionalChildProperties.properties as ComponentProperty[]).forEach((childProperty) => {
```

Comment:
Just wondering if this cast is safe when the property can also be a string? Didn't see any references to string based properties in the tests either?

Reply:
It is safe, the typing is incorrect. In `ParsedComponent`s these `properties` arrays have all been completely expanded with the definition from `componentProperties`

Reply:
In `ParsedComponent`, `properties` is defined as `properties: ComponentProperty[];` whereas in `Component` it is defined as `properties?: (string | ComponentProperty)[];`

`childProperties` is part ComponentProperty as `properties: (string | ComponentProperty)[]`. Once parsed it is really down to `properties: ComponentProperty[]`. That would require a `ParsedComponentProperty` type similar to ParsedComponent. I quickly tried to make that work but kept running into compile errors and decided to revert